catalog/core/catalog-core-transform-impl/src/test/java/org/codice/ddf/catalog/locator/impl/TransformerLocatorImplTest.java

```diff
@@ -0,0 +1,366 @@
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.locator.impl;
+
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThat;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Matchers.isA;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import com.google.common.collect.ImmutableMap;
+import ddf.catalog.Constants;
+import ddf.catalog.transform.InputTransformer;
+import ddf.catalog.transform.MetacardTransformer;
+import ddf.catalog.transform.QueryResponseTransformer;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import javax.activation.MimeType;
+import javax.activation.MimeTypeParseException;
+import org.codice.ddf.catalog.transform.MultiInputTransformer;
+import org.codice.ddf.catalog.transform.MultiMetacardTransformer;
+import org.codice.ddf.catalog.transform.TransformerProperties;
+import org.codice.ddf.catalog.transform.impl.InputTransformerAdapter;
+import org.codice.ddf.catalog.transform.impl.MetacardTransformerAdapter;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.stubbing.Answer;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.Filter;
+import org.osgi.framework.FrameworkUtil;
+import org.osgi.framework.InvalidSyntaxException;
+import org.osgi.framework.ServiceReference;
+
+@SuppressWarnings("unchecked")
+public class TransformerLocatorImplTest {
+
+  private BundleContext bundleContext;
+
+  private TransformerLocatorImpl transformerLocator;
+
+  /** Map of the mock service references registered for each class. */
+  private Map<Class, List<ServiceReference>> serviceMap;
+
+  @Before
+  public void setup() throws InvalidSyntaxException {
+    bundleContext = mock(BundleContext.class);
+
+    transformerLocator = new TransformerLocatorTest(bundleContext);
+
+    serviceMap = new HashMap<>();
+
+    when(bundleContext.getServiceReferences(isA(Class.class), anyString()))
+        .thenAnswer(
+            (Answer<Collection<ServiceReference>>)
+                invocationOnMock -> {
+                  Class<?> clazz = (Class<?>) invocationOnMock.getArguments()[0];
+                  String filter = (String) invocationOnMock.getArguments()[1];
+
+                  if (filter != null) {
+                    Filter f = FrameworkUtil.createFilter(filter);
+                    return serviceMap
+                        .get(clazz)
+                        .stream()
+                        .filter(f::match)
+                        .collect(Collectors.toList());
+                  }
+                  if (serviceMap.containsKey(clazz)) {
```

Comment:
If `filter` is null then you should just be able to return `serviceMap.get(clazz)` since that will return null if there's no mapping.